### ISSUE TITLE
Unable to find docker image

### ISSUE THREAD

**Issue by Phlegethon90** [freedomofpress/dangerzone#470]
Hello!

Running the app results in the following error:
```
docker: Error response from daemon: failed to resolve reference "dangerzone.rocks/dangerzone:latest": dangerzone.rocks/dangerzone:latest: not found.
```

How do I fix this?

**Comment by deeplow** [comment]
Hello! How are you running the application, if I may ask? Also, which operating system are you using? (windows, macOS, linux?)

Did you download it from the website or are you trying to build it yourself? If that's the case, you probably forgot the part of building the container image. Depending on the platform, you need to run something like this `./install/linux/build-image.sh`.

**Comment by Phlegethon90** [comment]
Cloned the repository and started `install/macos/build-image.sh`. Taking forever...

**Comment by deeplow** [comment]
Yes, it takes quite a bit. If you want to just get it working you can download the `.dmg` file for mac [from the downloads page](https://dangerzone.rocks/#downloads). We technically do release in homebrew, but we don't have as much testing. Hence why we don't mention it in our website.

**Comment by Phlegethon90** [comment]
Yup, release in homebrew is totally broken.

**Comment by deeplow** [comment]
It looks like it is. Sorry about that. We'll have to reevaluate our position on its support. We either support it officially and test it or we don't and don't lead users down wrong paths.

Our upcoming release (as .dmg) wil add initial support for update notifications. It's small step in the right directions towards keeping the software updated. It is my understanding that that's a strong reason for some using homebrew, so that could be something that you're looking for.

Sorry, again!

**Comment by apyrgio** [comment]
Hm, there's something in this bug report that reminds me of a bad error handling in Dangerzone when the disk was full: https://github.com/freedomofpress/dangerzone/issues/193

@Phlegethon90 thanks a lot for informing us about this error. If you have some time, I'd like us to dig into it more and understand the root cause, so that we can help the rest of our MacOS users:

1. Can you check your remaining disk space? E.g., from the terminal, you can run `df -h /`. If it's close to 100%, then we may have spotted the source of the error.

2. If you are comfortable with the MacOS terminal, can you give us the output of the following commands?


   ```
   docker images
   docker load -i /Applications/Dangerzone.app/Contents/MacOS/share/container.tar.gz
   ````
   
   (The first command outputs the Docker images, whereas the second command loads the Dangerzone image into Docker. It's basically what Dangerzone does under the hood the first time it starts up. Feel free to redact any sensitive info.)

**Comment by Phlegethon90** [comment]
1. Around 300 GB is free.
2. `docker images` does not show the tagged image. `docker load -i` results in `archive/tar: invalid tar header`.

**Comment by apyrgio** [comment]
Thanks for the output, we have a lead then!

Can you provide us with the Docker version (`docker version`)?

**Comment by Phlegethon90** [comment]
```
Client:
 Cloud integration: v1.0.35
 Version:           24.0.2
 API version:       1.43
 Go version:        go1.20.4
 Git commit:        cb74dfc
 Built:             Thu May 25 21:51:16 2023
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.21.1 (114176)
 Engine:
  Version:          24.0.2-38-g8e70a1b23e
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.4
  Git commit:       8e70a1b23e965d86ec8c2feb77605196ae124630
  Built:            Fri Jun  2 15:58:50 2023
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

```
-rw-r--r--@ 1 <local username>  admin  700786238 19 Apr 20:31 /Applications/Dangerzone.app/Contents/MacOS/share/container.tar.gz
```

**Comment by apyrgio** [comment]
Hm, that's interesting. In my machine, I have the exact same Docker version as you, and the `container.tar.gz` file has the same size and permissions. And yet, I can load it with `docker load -i <container.tar.gz>` normally.

Let's make sure that there's no corruption in the image. In my case, I see the following hash:

```console
$ shasum -a 256 /Applications/Dangerzone.app/Contents/MacOS/share/container.tar.gz
ab017a4209e4e6e664a8d3a5d0108ead8335389d7bdb288cb72765af558a20e4  /Applications/Dangerzone.app/Contents/MacOS/share/container.tar.gz
```

Do you see the same result as well?

**Comment by apyrgio** [comment]
Hi @Phlegethon90. Gentle ping for this one, do you have any update here? We also released a newer Dangerzone version, have you perhaps tried it out and did it resolve your issue?

**Comment by harrislapiroff** [comment]
Closing this since we weren't able to reproduce. The next version of Dangerzone will not depend on Docker, so we expect any issue here to be resolved naturally. (If you have a way to reproduce, feel free to comment and we can reopen and consider solving it sooner.)
